### PR TITLE
lumo 1.9.0 -> 1.10.1 plus darwin support

### PR DESCRIPTION
**Pull request by hlolli**
###### Motivation for this change
New release, also some requests for adding darwin support which I implemented by adding two missing buildDependencies.

###### Things done
Removed and added some npm packages, rebuilt the node10 packages files. Tested in on my linux and on MacOs Sierra in VirtualBox.

- [ x ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ x ] NixOS
   - [ x ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ x ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by hlolli** [comment]
rebased and merge conflict resolved. Can this be merged @markuskowa ?

**Comment by Mic92** [comment]
I will have a look, once it gets updated.

**Comment by raboof** [comment]
I ran into #79781 and can confirm that when using lumo from this branch it does work.

Seems like this PR needs a rebase, though?

**Comment by hlolli** [comment]
Because of not being rebased, it will work. I'm finishing work now. I think pkgs.clojure needs a patch or older version provided. Will update in ~2 hours.

**Comment by hlolli** [comment]
the branch is rebased now, but and the bug is there. Will try few things out, starting with simplest solutions.

**Comment by hlolli** [comment]
The problem seems to be purely due to clojure, adding this check to interpreters/clojure/default.nix

```
doInstallCheck = true;
  installCheckPhase = ''
   CLJ_CONFIG=`pwd` CLJ_CACHE=$tmp $out/bin/clojure -Spath
  '';
```

causes

```
Error building classpath. Failed to read artifact descriptor for org.clojure:clojure:jar:1.10.1
org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.clojure:clojure:jar:1.10.1
```

I think the best way to solve it is by providing the clojure binary an environment where it has access to all its runtime jar deps. This is easy to do, and they do provide deps.edn which clojure reads from on each invocation.

**Comment by hlolli** [comment]
after https://github.com/NixOS/nixpkgs/pull/79868 gets merged, Lumo should work fine, with or without this bump (tough I'd love to see this merged since it's been stale for almost a year).

File: pkgs/development/node-packages/node-env.nix [hunk 1]
```diff
@@ -363,7 +354,7 @@ let
 
         npm ${forceOfflineFlag} --nodedir=${nodeSources} ${npmFlags} ${stdenv.lib.optionalString production "--production"} rebuild
 
-        if [ "''${dontNpmInstall-}" != "1" ]
+        if [ "$dontNpmInstall" != "1" ]
```
Comment (Mic92) [inline pkgs/development/node-packages/node-env.nix]: This should stay!

File: pkgs/development/node-packages/composition-v10.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This file has been generated by node2nix 1.7.0. Do not edit!
+# This file has been generated by node2nix 1.6.1. Do not edit!
```
Comment (Mic92) [inline pkgs/development/node-packages/composition-v10.nix]: It is weird that this was now regenerated using an older version of node2nix. Did you generated it from an up-to-date nixpkgs repository?

Comment (hlolli) [reply pkgs/development/node-packages/composition-v10.nix]: ah, yes I have older version managed by npm -g, I should ofc install it via nix (but there were some bugs fixed upstream that I needed for something sometime).

Will fix after work

File: pkgs/development/node-packages/node-env.nix (hunk 1 above)
Comment (hlolli) [reply pkgs/development/node-packages/node-env.nix]: I have no recollection from this change. Most likely the generated data from 1.6.1. You mean this should stay, the change or the original? (I assume the original, in which case regenerating with right version should fix it).

Comment (Mic92) [reply pkgs/development/node-packages/node-env.nix]: It is a bug in node2nix itself: https://github.com/svanderburg/node2nix/pull/174

Comment (Mic92) [reply pkgs/development/node-packages/node-env.nix]: We need to keep the current line.

**Comment by hlolli** [comment]
@Mic92 I finally got around fixing the comments, the build was also failing, for the same reason the current Lumo build is failing. So I added 2 extra patches and the build goes trough. It would be nice if this could ship with the next release, because as far as I see, Lumo would otherwise get a broken status. I guess for that I'd need to change base?

**Comment by raboof** [comment]
branch is still working for me :smile: :+1:

**Pull request merged**